metricflow/dataflow/dataflow_plan.py

```diff
@@ -791,13 +799,13 @@ def functionally_identical(self, other_node: DataflowPlanNode[SourceDataSetT]) -
     def with_new_parents(  # noqa: D
         self, new_parent_nodes: Sequence[BaseOutput[SourceDataSetT]]
     ) -> JoinToTimeSpineNode[SourceDataSetT]:
-        # Is this right? how is this method used?
         assert len(new_parent_nodes) == 1
         return JoinToTimeSpineNode[SourceDataSetT](
             parent_node=new_parent_nodes[0],
+            metric_time_dimension_specs=self.metric_time_dimension_specs,
```

Comment:
We need to add a check to the `functionally_identical` method to ensure the metric_time_dimension_specs are the same between other_node and self. 

These methods are used for an impressively difficult-to-reason-about optimizer that prunes out part of the dataflow plan, so if we miss that equivalence check we run the risk of removing a subquery that we need to remain in place.

Reply:
Good catch!! Updated!